ichnaea/api/exceptions.py

```diff
@@ -152,6 +152,17 @@ class ParseError(BaseAPIClientError):
     domain = "global"
     reason = "parseError"
     message = "Parse Error"
+    error_details = None
+
+    def __init__(self, details=None):
+        self.error_details = details
+        super().__init__()
+
+    def json_body(self):
```

Comment:
This is shadowing a class method with an instance method. That's problematic. I think it's probably better to make `json_body` an instance method in all these classes and adjust the tests accordingly.

Reply:
Probably worth fixing `ParseError` which is also shadowing a class method with an instance method.

Reply:
There's a lot of code that relies on ``json_body`` being a class method, but I think most of it is test code. I'll give this a shot.